next/test/fields/visibility.test.ts

```diff
@@ -214,5 +213,45 @@ describe('Field visibility', () => {
       } })
       expect(getField(form.fields, 'form', 'password')?.isVisible).toBe(true)
     })
+
+    it('should allow hiding the whole fieldset', () => {
+      const conditionalFieldsetSchema: JsfObjectSchema = {
+        type: 'object',
+        properties: {
+          hide_form: {
+            type: 'string',
+          },
+          form: {
+            type: 'object',
+            properties: {
+              name: {
+                type: 'string',
+              },
+              password: {
+                type: 'string',
+              },
+            },
+          },
+        },
+        if: {
+          properties: {
+            hide_form: {
+              const: 'yes',
+            },
+          },
+          required: ['hide_form'],
+        },
+        then: {
+          properties: {
+            form: false,
+          },
+        },
+      }
+
+      const form = createHeadlessForm(conditionalFieldsetSchema, { initialValues: { form: { name: 'admin', password: null } } })
+      form.handleValidation({ hide_form: 'yes' })
+
+      expect(getField(form.fields, 'form')?.isVisible).toBe(false)
```

Comment:
q: If we check `form.name`, will it also have `isVisible: false`? No real bug comes to my mind, just wondering.